### DISCUSSION TITLE
Stepping motor skip or does not rotate in the expected direction when using Web Shifter using a NonPCB configuration

### DISCUSSION THREAD

**Discussion by evanscastonguay** [doudar/SmartSpin2k#419] [Q&A]
**Summary**
when I try to change gear using the Web Shifter the steeping motor is skipping or not rotating the expected direction. In fact it is randomly going in different direction.

**Background**
I have successfully 
1. purchase all the parts to build my smartspin2k using a non PCB configuration.
2. upload the smartspin2k firmware software in the ESP32 
3. setup all smartspin2k components using a non PCB configuration. (power supply 12v, 5v buckconverter, ESP32, tmc2225, Nema step motor) see ![IMG_0688](https://user-images.githubusercontent.com/8137725/208725258-e35738f3-d33a-427a-ba8a-3db53807a337.JPG).Note: at this point I am not connecting any Pushbuttons hardware on the ESP32 to change gear. I assume this shall be ok.
4. connect to the ESP32 using wifi and configure the ESP32 to use my local network 
5. Use the Web Shifter page to simulate changing gear and observe if the motor is rotating in the right direction and rotating smottly

**Current behavior**
The stepping motor is skipping or rotating in the wrong direction randomly. see video https://youtu.be/F-sT1P2Qq_k
During the first 25 seconds of the video I move the gear up and the motor rotate clockwise. Then I start moving the gear down but the motor keep moving clockwise. Also the motor seam to skip and I am not sure this is normal.

**Expectated behavior**
The stepping motor shall not skip and when we increase the gear up the motor shall rotate in one direction and when gear down it shall rotate in the other direction.

**Comment by doudar** [comment]
To me, the suspect area would be a bad connection on the dir pin. You could try hooking that pin up to 3.3v temporarily and see if it fixes the issue. 

Otherwise possibly you have two wires reversed (like dir and enable), or one of the stepper output wires needs to be moved. I've seen plenty of stepper wire cables that weren't correctly paired.

**Reply by doudar** [reply]
Actually, looking at the picture, it appears you might have the en pin connected to uart? If so you'll need it connect to at least 3.3v or the appropriate pin on the esp32 (gpio 13)

Edit: On second look it appears en is connected properly.

Edit Edit 😄 I see you crossed your stepper cables on the extension you made to the breadboard. The wires in the stepper cable supplied with the motor are already crossed so you need to just connect them straight through just like if you plugged the jst-xh cable directly into the stepper driver.

**Reply by evanscastonguay** [reply]
Thank this fix the problem. The issue was indeed that my stepper cable was not connect correctly. After I switch back the connection and retest all my connection the motor work smoothly without skipping. Thank you for your support.
